Source/Core/Common/Subscribable.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+
+#include "Common/NonCopyable.h"
+
+template <typename... Args>
+class Subscribable : NonCopyable
```

Comment:
Done. I added some basic functionality tests (sending a value to no subscribers, some subscribers, sending multiple times, removing subscribers).